simulator/infra/ports/module.h

```diff
@@ -21,17 +23,22 @@ class Module : public Log
     template<typename T>
     auto make_write_port( std::string key, uint32 bandwidth) 
     {
+        topology_write_ports.push_back(key);
         return std::make_unique<WritePort<T>>( get_portmap(), std::move(key), bandwidth);
     }
 
     template<typename T>
     auto make_read_port( std::string key, Latency latency)
     {
+        topology_read_ports.push_back(key);
         return std::make_unique<ReadPort<T>>( get_portmap(), std::move(key), latency);
     }
 
     void enable_logging_impl( const std::unordered_set<std::string>& names);
 
+    void module_dumping( boost::property_tree::ptree* modules) const;
+    void modulemap_dumping( boost::property_tree::ptree* modulemap) const;
```

Comment:
Can this method be private?

Reply:
They must be protected for testing.

Reply:
No, such test would not be correct.
Class internals are not considered to be exposed. For instance, if I'd like to rename the function, or split it in two, I would expect that class customers do not see the change.